lib/ansible/modules/cloud/amazon/ec2_lc.py

```diff
@@ -151,164 +174,280 @@
 
 try:
-    from boto.ec2.blockdevicemapping import BlockDeviceType, BlockDeviceMapping
-    import boto.ec2.autoscale
-    from boto.ec2.autoscale import LaunchConfiguration
-    from boto.exception import BotoServerError
-    HAS_BOTO = True
-except ImportError:
-    HAS_BOTO = False
-
-
-def create_block_device(module, volume):
-    # Not aware of a way to determine this programatically
-    # http://aws.amazon.com/about-aws/whats-new/2013/10/09/ebs-provisioned-iops-maximum-iops-gb-ratio-increased-to-30-1/
-    MAX_IOPS_TO_SIZE_RATIO = 30
-    if 'snapshot' not in volume and 'ephemeral' not in volume:
-        if 'volume_size' not in volume:
-            module.fail_json(msg='Size must be specified when creating a new volume or modifying the root volume')
-    if 'snapshot' in volume:
-        if 'device_type' in volume and volume.get('device_type') == 'io1' and 'iops' not in volume:
-            module.fail_json(msg='io1 volumes must have an iops value set')
-    if 'ephemeral' in volume:
-        if 'snapshot' in volume:
-            module.fail_json(msg='Cannot set both ephemeral and snapshot')
-    return BlockDeviceType(snapshot_id=volume.get('snapshot'),
-                           ephemeral_name=volume.get('ephemeral'),
-                           size=volume.get('volume_size'),
-                           volume_type=volume.get('device_type'),
-                           delete_on_termination=volume.get('delete_on_termination', False),
-                           iops=volume.get('iops'),
-                           encrypted=volume.get('encrypted',None))
-
-
-def create_launch_config(connection, module):
-    name = module.params.get('name')
-    image_id = module.params.get('image_id')
-    key_name = module.params.get('key_name')
-    vpc_id = module.params.get('vpc_id')
-    try:
-        security_groups = get_ec2_security_group_ids_from_names(module.params.get('security_groups'), ec2_connect(module), vpc_id=vpc_id, boto3=False)
-    except ValueError as e:
-        module.fail_json(msg=str(e))
-    user_data = module.params.get('user_data')
-    user_data_path = module.params.get('user_data_path')
-    volumes = module.params['volumes']
-    instance_type = module.params.get('instance_type')
-    spot_price = module.params.get('spot_price')
-    instance_monitoring = module.params.get('instance_monitoring')
-    assign_public_ip = module.params.get('assign_public_ip')
-    kernel_id = module.params.get('kernel_id')
-    ramdisk_id = module.params.get('ramdisk_id')
-    instance_profile_name = module.params.get('instance_profile_name')
-    ebs_optimized = module.params.get('ebs_optimized')
-    classic_link_vpc_id = module.params.get('classic_link_vpc_id')
-    classic_link_vpc_security_groups = module.params.get('classic_link_vpc_security_groups')
-    bdm = BlockDeviceMapping()
-
-    if user_data_path:
+    import botocore
+except Exception:
+    pass
+
+
+class Ec2LaunchConfigurationServiceManager(object):
```

Comment:
I'm not at all convinced this class is useful, it seems to just makes the code flow harder to read. I'd also think that the diff could be reduced without the change to the class structure, which would also make this PR easier to review.